cockatrice/src/window_main.cpp

```diff
@@ -431,46 +431,59 @@ QString MainWindow::extractInvalidUsernameMessage(QString & in)
     return out;
 }
 
-void MainWindow::registerError(Response::ResponseCode r, QString reasonStr, quint32 endTime)
+void MainWindow::registerError(Response::ResponseCode r, QString reasonStr, quint32 endTime, QString customStr)
 {
+    qDebug() << "RECEIVED RESPONSE!";
+    QString titleStr = "Registration";
+    QString explinationStr = "";
     switch (r) {
         case Response::RespRegistrationDisabled:
-            QMessageBox::critical(this, tr("Registration denied"), tr("Registration is currently disabled on this server"));
+            titleStr = titleStr + " denied";
```

Comment:
This breaks the translation. The whole string to be translated needs to be in a `tr()` textually.

So what we want to do is actually keep the full `tr` calls in each switch case, and then *after* the switch case, append `\n\n` and the untranslated custom message. So like

```cpp
QString titleStr;
QString detailStr;
switch (r) {
  case Response::RespRegistrationDisabled:
    titleStr = tr("Registration denied");
    detailStr = tr("Registration is currently disabled on this server");
    break;
  case ...
  case default: ...
}
if (!customStr.isEmpty)
  detailStr += "\n\n" + tr("Note from server: ") + customStr;
QMessageBox::critical(this, titleStr, detailStr);
```